packages/publisher/src/publish-packages.ts

```diff
@@ -187,6 +192,85 @@ export default async function publishPackages(
     cacheDirPath
   );
 
+  // Loop over the @types packages in npm and mark any that no longer
+  // exist in HEAD as deprecated.
+  let from = 0;
+  let objects;
+  do {
+    const opts = {
+      limit: 250,
+      from
+    };
+    objects = await search("@types", opts);
+    for (const { name: fullNpmName } of objects) {
+      const name = fullNpmName.slice("@types/".length);
+      // If they don't exist in the types directory or in
+      // notNeededPackages.json then mark them deprecated. Reference the
+      // commit/pull request that removed them.
+      if (!allPackages.tryGetLatestVersion(name) && !allPackages.getNotNeededPackage(name)) {
```

Comment:
probably don't want to waste time deprecating already-deprecated things either.

Reply:
I think the query takes care of this: It doesn't return deprecated results.